src/window/src/no_window.rs

```diff
@@ -0,0 +1,95 @@
+
+//! An implementation of Window that runs without a window at all.
+//!
+//! It saves just enough information to implement the window interface,
+//! but otherwise does nothing.
+
+extern crate input;
+
+use {
+	Window,
+	WindowSettings,
+	BuildFromWindowSettings,
+	AdvancedWindow
+};
+
+use input::Input;
+
+/// The tiny NoWindow structure.
+///
+/// This structure holds just enough state to be able to return
+/// values that were set.
+pub struct NoWindow {
+    should_close: bool,
+    title: String,
+    size: Size
+}
+
+impl NoWindow {
+    /// Returns a new `NoWindow`.
+    pub fn new(settings: WindowSettings) -> NoWindow {
+        NoWindow {
+            should_close: false,
+            title: settings.title,
+            size: settings.size,
+        }
+    }
+}
+
+impl Window for NoWindow {
+	/// Uses the usual event type from the input module.
+    type Event = Input;
+    
+    /// Gets the stored value for `should_close`.
+    fn should_close(&self) -> bool { self.should_close }
+    
+    /// Saves a new value for `should_close`.
+    ///
+    /// Otherwise a no-op.
```

Comment:
Can remove "Otherwise a no-op" since other window backends also use it as a flag, while the actual cleaning up of the window runs in the destructor of the window object. I think the sentence is unnecessarily as the behavior is very similar to other backends.